src/Castle.Core/DynamicProxy/CustomAttributeInfo.cs

```diff
@@ -34,26 +36,26 @@ public class CustomAttributeInfo : IEquatable<CustomAttributeInfo>
 		// Cached empty arrays to avoid unnecessary allocations
 		private static readonly PropertyInfo[] EmptyProperties = new PropertyInfo[0];
 		private static readonly FieldInfo[] EmptyFields = new FieldInfo[0];
-		private static readonly object[] EmptyValues = new object[0];
+		private static readonly object?[] EmptyValues = new object?[0];
 
-		private static readonly IEqualityComparer<object> ValueComparer = new AttributeArgumentValueEqualityComparer();
+		private static readonly AttributeArgumentValueEqualityComparer ValueComparer = new AttributeArgumentValueEqualityComparer();
 
 		private readonly CustomAttributeBuilder builder;
 		private readonly ConstructorInfo constructor;
-		private readonly object[] constructorArgs;
-		private readonly IDictionary<string, object> properties;
-		private readonly IDictionary<string, object> fields;
+		private readonly object?[] constructorArgs;
+		private readonly IDictionary<string, object?> properties;
+		private readonly IDictionary<string, object?> fields;
 
 		public CustomAttributeInfo(
 			ConstructorInfo constructor,
-			object[] constructorArgs,
+			object?[] constructorArgs,
 			PropertyInfo[] namedProperties,
-			object[] propertyValues,
+			object?[] propertyValues,
 			FieldInfo[] namedFields,
-			object[] fieldValues)
+			object?[] fieldValues)
 		{
 			// Will take care of validating the arguments
-			this.builder = new CustomAttributeBuilder(constructor, constructorArgs, namedProperties, propertyValues, namedFields, fieldValues);
+			this.builder = new CustomAttributeBuilder(constructor, constructorArgs, namedProperties, propertyValues!, namedFields, fieldValues!);
```

Comment:
I should verify whether `CustomAttributeBuilder` really does not accept `null` values here (or whether, like I thought, their nullable reference type annotation isn't entirely correct).

Reply:
Verified. The annotation was not correct in `netstandard2.0` but fixed later on. So for that TFM we indeed need the null-forgiving `!`.